state/aerospike/aerospike.go

```diff
@@ -162,9 +162,9 @@ func (aspike *Aerospike) Get(req *state.GetRequest) (*state.GetResponse, error)
 
 	policy := &as.BasePolicy{}
 	if req.Options.Consistency == state.Strong {
-		policy.ConsistencyLevel = as.CONSISTENCY_ALL
+		policy.ReadModeAP = as.ReadModeAPAll
```

Comment:
From the code: https://github.com/aerospike/aerospike-client-go/blob/v4.5.0/policy.go#L53
It seems there are two parameters to consider? Is that the scenario?
```
// ReadModeAP indicates read policy for AP (availability) namespaces.
ReadModeAP ReadModeAP //= ONE

// ReadModeSC indicates read policy for SC (strong consistency) namespaces.
ReadModeSC ReadModeSC //= SESSION;
```
The same file in v2.7.0 https://github.com/aerospike/aerospike-client-go/blob/v2.7.0/policy.go#L47
```
// How replicas should be consulted in a read operation to provide the desired
// consistency guarantee. Default to allowing one replica to be used in the
// read operation.
ConsistencyLevel ConsistencyLevel //= CONSISTENCY_ONE
```

Reply:
I saw that, and I read:
https://github.com/aerospike/aerospike-client-go/blob/master/read_mode_sc.go

My interpretation was that SESSION is an ok default, since it means the clients only read from the master server in this case? If someone with more familiarity with the upstream service wants to make the case that this should be ReadModeSCLinearize, I can change it... This seems to me like it would only matter if this resource was being shared with non-dapr clients?

Reply:
I did some more digging:
https://aerospike.com/blog/developers-understanding-aerospike-transactions/

I'm not sure we have the same definition of strong consistency that they do? For us strong consistency means that all replicas have committed a change before a write operation completes. For them, it seems like consistency is enforced on the read side? But the more I read, the more I think we probably want ReadModeSCLinearize, here.

Good call out, I will change this.

Reply:
I wonder if @abhirockzz has any knowledge of this.